src/test/java/github/grace000/erversay/ServerConnectionTest.java

```diff
@@ -0,0 +1,38 @@
+package github.grace000.erversay;
+
+import org.junit.Test;
+import static org.junit.Assert.*;
+import java.io.IOException;
+
+
+public class ServerConnectionTest {
+   private MockConnection mockConnection = new MockConnection();
+   private String response = "test";
+
+
+    @Test
+    public void itListensForAConnection() {
+        int port = 1234;
+        mockConnection.listenForConnection(port);
+        assertTrue(mockConnection.isListenForConnectionCalled());
```

Comment:
Does this test anything outside your mock? Or just if your mock works?

Reply:
Apply this question to every one of the tests in this file, please.

Reply:
I have changed the file so that I'm using a mock sockets with the actual connection class instead.